CMakeLists.txt

```diff
@@ -51,14 +52,21 @@ set(_dpctl_sycl_target_compile_options)
 set(_dpctl_sycl_target_link_options)
 
 set(_dpctl_sycl_targets)
+set(_dpctl_cuda_arch)
 set(_dpctl_amd_targets)
 if ("x${DPCTL_SYCL_TARGETS}" STREQUAL "x")
-   if (DPCTL_TARGET_CUDA)
-      set(_dpctl_sycl_targets "nvptx64-nvidia-cuda,spir64-unknown-unknown")
-   else()
-      if (DEFINED ENV{DPCTL_TARGET_CUDA})
-          set(_dpctl_sycl_targets "nvptx64-nvidia-cuda,spir64-unknown-unknown")
+   if (NOT "x${DPCTL_TARGET_CUDA}" STREQUAL "x")
+      if (DPCTL_TARGET_CUDA STREQUAL "ON")
```

Comment:
I think this can be made a bit more robust, per [CMake documentation](https://cmake.org/cmake/help/latest/command/if.html#constant)
```
True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number
```
then for `if(<variable>)`
```
True if given a variable that is defined to a value that is not a false constant.
```
if this is accurate, then `if(DPCTL_TARGET_CUDA)` can be used to check for true/false, and if true, set to default value, then we can check for a match with `"sm_*"`

then we can also allow `DPCTL_TARGET_CUDA=FALSE` or `=0`, etc. and just pass if given this way

Reply:
CMake treats any non-false value as `True` including invalid inputs such as `qwe`. This will incorrectly set the default value to `sm_50`

To handle this correctly we still have to explicitly check that `DPCTL_TARGET_CUDA MATCHES “^(ON|TRUE|YES|1)$”` before assigning default value and throw `FATAL_ERROR in `else` branch for invalid inputs.

If it is ok I will implement it

Reply:
done